lib/src/checks.dart

```diff
@@ -1,4 +1,48 @@
+import 'package:nyxx/nyxx.dart';
 import 'package:nyxx_commands/nyxx_commands.dart';
+import 'package:running_on_dart/src/services/music.dart';
 import 'package:running_on_dart/src/settings.dart';
 
 final administratorCheck = UserCheck.anyId(adminIds, 'Administrator check');
+
+final connectedToAVoiceChannelCheck = Check((IContext context) async {
+  final selfMember = await context.guild!.selfMember.getOrDownload();
+
+  if (selfMember.voiceState == null || selfMember.voiceState!.channel == null) {
+    throw MusicCheckException(context, 'I have to be in a voice channel to use this command');
+  }
+  return true;
+});
+
+final notConnectedToAVoiceChannelCheck = Check((IContext context) async {
+  final selfMember = await context.guild!.selfMember.getOrDownload();
+
+  if (selfMember.voiceState == null || selfMember.voiceState!.channel == null) {
+    return true;
+  }
+  throw MusicCheckException(context, "I'm already connected to a voice channel");
+});
+
+final sameVoiceChannelOrDisconnectedCheck = Check((IContext context) async {
+  // If this is an interaction, acknowledge it just in case the check
+  // takes too long to run.
+  if (context is InteractionChatContext) {
+    await context.acknowledge();
```

Comment:
No need to manually acknowledge the interaction, `nyxx_commands` will do it for us if the command takes too long to execute.

Reply:
it doesn't seem to, that's why i'm acknowledging it manually

Reply:
Hmm, let me try. It might be an issue with nyxx_commands and nyxx_interactions being inflexible when it comes to latency/client-server clock desync.

Reply:
It seems to work fine for me. We can leave it in for now and remove it later when nyxx_commands updates to fix it.